app/main.py

```diff
@@ -2,5 +2,16 @@
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_results = {}
+
+    def inner(*args) -> int:
+        if args not in cached_results:
+            print("Calculating new result")
+            result = func(*args)
+            cached_results[args] = result
+            return result
```

Comment:
try to rewrite your decorator not to have `return` duplicated

Reply:
@fsocie7y like the push i added last?

Reply:
Do i have to get rid of "else" block somehow, or it is fine?